### PR TITLE
[SPARK-23539][SS][FOLLOWUP][TESTS] Add UT to ensure existing query doesn't break with default conf of includeHeaders

### PR DESCRIPTION
**Pull request by HeartSaVioR**
### What changes were proposed in this pull request?

This patch adds new UT to ensure existing query (before Spark 3.0.0) with checkpoint doesn't break with default configuration of "includeHeaders" being introduced via SPARK-23539.

This patch also modifies existing test which checks type of columns to also check headers column as well.

### Why are the changes needed?

The patch adds missing tests which guarantees backward compatibility of the change of SPARK-23539.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

UT passed.

### CODE REVIEW

**Comment by HeartSaVioR** [comment]
This patch is to address my own comment which PR was merged without addressing this.
https://github.com/apache/spark/pull/22282#issuecomment-476883917

**Comment by HeartSaVioR** [comment]
Actually the other intention was let test fail when "includeHeaders" is set to "true", as we suspected from https://github.com/apache/spark/pull/22282#issuecomment-463008868, but turns out it doesn't break even the option is set to true. It's because of the mechanism how `dropDuplicate` works - row is stored as key, and getting value via key doesn't leverage type while comparing as both hashCode and equals of row don't use type.

Luckily it doesn't let query crash, but it would be ideal to do type check and avoid this case (possible chance to crash in runtime) - SPARK-27237 (#24173) has been proposed to address this.

**Comment by HeartSaVioR** [comment]
cc. @zsxwing @srowen as reviewers with commit privileges of #24173 , and @dongjinleekr as author of #24173

**Review by srowen — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]
seems fine to me.

File: external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchSourceSuite.scala [hunk 1]
```diff
@@ -1162,6 +1164,62 @@ class KafkaMicroBatchV2SourceSuite extends KafkaMicroBatchSourceSuiteBase {
     intercept[IllegalArgumentException] { test(minPartitions = "-1", 1, true) }
   }
 
+  test("default config of includeHeader doesn't break existing query from Spark 2.4") {
```
Comment (dongjinleekr) [inline external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchSourceSuite.scala]: +1 for this test. :smile:

**Comment by HeartSaVioR** [comment]
FYI, CI build just got passed.

**Comment by srowen** [comment]
Merged to master

**Comment by HeartSaVioR** [comment]
Thanks all for reviewing and merging!

**Pull request closed without merging**